validation/serve.py

```diff
@@ -18,36 +20,43 @@ def get_args():
 app = FastAPI()
 args, _ = get_args()
 
-#################################
-# For testing
-#################################
-# loader = HDF5Loader.HDF5Loader()
-# pcl_data = loader.load_point_cloud_from_h5("bear", "/home/tesha/Documents/Python/404_dream_gaussian/mining/logs/output8")
-# pcl_buffer = loader.pack_point_cloud_to_io_buffer(pcl_data['points'], pcl_data['normals'], pcl_data['features_dc'], pcl_data['features_rest'],
-#                                                   pcl_data['opacities'], pcl_data['scale'], pcl_data['rotation'], pcl_data['sh_degree'])
-# pcl_buffer_encode = base64.b64encode(pcl_buffer.getbuffer())
-#
-# prompt = "an image of the bear"
-# message = TextTo3D(prompt_in=prompt, mesh_out=pcl_buffer_encode)
-
-
-@app.get("/validate")
-# def validate_gs_result(save_images: bool = False):
-def validate_gs_result(message: TextTo3D, save_images: bool = False):
-    print("[INFO] Start validating the input 3D data.")
-    print("[INFO] Input prompt: ", message.prompt_in)
 
+class RequestData(BaseModel):
+    prompt: str
+    data: str
+
+
+class ResponseData(BaseModel):
+    scores: List[float]
+
+
+@app.post("/validate/", response_model=ResponseData)
+async def validate(request: RequestData) -> ResponseData:
+    """
+    Validates the input prompt and data to produce scores.
+
+    Parameters:
+    - request (RequestData): An instance of RequestData containing the input prompt and data.
+
+    Returns:
+    - ResponseData: An instance of ResponseData containing the scores generated from the validation process.
+
+    """
+    print("[INFO] Start validating the input 3D data.")
+    print("[INFO] Input prompt: ", request.prompt)
     t1 = time()
 
-    Validator = validation.ValidateTextTo3DModel(512, 512, 10)
-    Validator.init_gaussian_splatting_renderer()
-    scores = Validator.score_response_gs_input([message], save_images=save_images, cam_rad=4)
+    t23D = protocol.TextTo3D(prompt_in=request.prompt, mesh_out=request.data)
+    validator = TextTo3DModelValidator(512, 512, 10)
+    validator.init_gaussian_splatting_renderer()
+    scores = validator.score_response_gs_input([t23D], save_images=False, cam_rad=4)
 
     t2 = time()
     print("[INFO] Score: ", scores)
     print("[INFO] Validation took: ", t2 - t1, " sec")
```

Comment:
I'd prefer:
f"[INFO] Validation took: {t2 - t1} sec"

Reply:
Done. Changed it for all three print statements that took multiple arguments to keep things uniform.

Reply:
Also I would use a logger instead of prints for this.